pkg/controller/blockdeviceclaim/blockdeviceclaim_controller.go

```diff
@@ -193,7 +193,11 @@ func (r *ReconcileBlockDeviceClaim) FinalizerHandling(
 		return nil
 	}
 	// The object is being deleted
-	if util.Contains(instance.ObjectMeta.Finalizers, BlockDeviceClaimFinalizer) {
+	// Check if the BDC has only NDM finalizer present. If yes, it means that the BDC
+	// was deleted by the owner itself, and NDM can proceed with releasing the BD and
+	// removing the NDM finalizer
+	if len(instance.ObjectMeta.Finalizers) == 1 &&
+		util.Contains(instance.ObjectMeta.Finalizers, BlockDeviceClaimFinalizer) {
```

Comment:
Will this impact BDC's that were created with the older finalizer string in 1.0? Should we as part of upgrade patch all the BDCs with this name change or handle the backward compatibility in the code?

Reply:
Yes @kmova . We will  need to patch the finalizer as part of upgrade script. I have included this as a part of NDM upgrade steps.

Reply:
Also, one more query - in case of manual BDC creation by user. Can there be a case of nil failure if there are no finalizers.

Reply:
If there are no finalizers, then the `ObjectMeta.Finalizer` array will be empty. Therefore, 
`len(instance.ObjectMeta.Finalizers)` will be equal to zero.